ios/Classes/SwiftWorkmanagerPlugin.swift

```diff
@@ -1,14 +1,135 @@
 import Flutter
 import UIKit
+import os
 
-public class SwiftWorkmanagerPlugin: NSObject, FlutterPlugin {
-  public static func register(with registrar: FlutterPluginRegistrar) {
-    let channel = FlutterMethodChannel(name: "workmanager", binaryMessenger: registrar.messenger())
-    let instance = SwiftWorkmanagerPlugin()
-    registrar.addMethodCallDelegate(instance, channel: channel)
-  }
+public class SwiftWorkmanagerPlugin: FlutterPluginAppLifeCycleDelegate {
+    
+    private struct ForegroundMethodChannel {
+        static let channelName = "be.tramckrijte.workmanager/foreground_channel_work_manager"
+        enum methods: String {
+            case initialize
+            case performSampleFetch
+        }
+    }
+    
+    private struct BackgroundMethodChannel {
+        static let channelName = "be.tramckrijte.workmanager/background_channel_work_manager"
+        enum methods: String {
+            case backgroundJobDidComplete
+        }
+    }
+    
+    private let flutterThreadLabelPrefix = "BackgroundFetch"
+    
+}
+
+//MARK: - FlutterPlugin conformance
+
+extension SwiftWorkmanagerPlugin: FlutterPlugin {
+    
+    public static func register(with registrar: FlutterPluginRegistrar) {
+        let foregroundMethodChannel = FlutterMethodChannel(name: ForegroundMethodChannel.channelName, binaryMessenger: registrar.messenger())
+        let instance = SwiftWorkmanagerPlugin()
+        registrar.addMethodCallDelegate(instance, channel: foregroundMethodChannel)
+        registrar.addApplicationDelegate(instance)
+    }
+    
+    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+        
+        switch call.method {
+        case ForegroundMethodChannel.methods.initialize.rawValue:
+            guard let callbackHandle: Int64 = call.arguments as? Int64 else {
+                result(WMPError.unexpectedMethodArguments(argumentsDescription: call.arguments.debugDescription))
+                return
+            }
+            store(callbackHandle)
+            result(nil)
+        case ForegroundMethodChannel.methods.performSampleFetch.rawValue:
+            triggerSampleBackgroundFetch()
+            result(nil)
+        default:
+            result(WMPError.unhandledMethod(methodName: call.method).asFlutterError)
+            return
+        }
+        
+    }
+    
+}
+
+//MARK: - AppDelegate conformance
+
+extension SwiftWorkmanagerPlugin {
+    
+    override public func application(_ application: UIApplication, performFetchWithCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) -> Bool {
+        
+        // First, let's retreive our callBack handle
+        guard let callbackHandle: Int64 = getStoredCallbackHandle() else {
+            completionHandler(.failed)
```

Comment:
I would go for the most optimistic approach here. Just forcefully unwrap it. If it crashes then fine. It should.
Now we are just swallowing that error.